coalib/results/SourceRange.py

```diff
@@ -10,7 +10,10 @@ class SourceRange(TextRange):
     @enforce_signature
     def __init__(self,
                  start: SourcePosition,
-                 end: (SourcePosition, None)=None):
+                 end: (SourcePosition, None)=None,
+                 text=None,
+                 start_position=None,
```

Comment:
Also, we seem to be using enforce_signature. SO you should do `start_position: (int, None)=None`